bootstrap_tarball.sh

```diff
@@ -1,168 +1,204 @@
-#!/bin/sh -e
+#!/bin/bash -e
 
-mkdir -p build/tarball build/stage1/bin build/stage1/lib/autogen build/prefix
 PREFIX="$(pwd)/build/stage1"
 FINALPREFIX="${FINALPREFIX:-$(pwd)/build/prefix}"
 GUILE_VERSION="${GUILE_VERSION:-3.0}"
 
-## Get the tarball, remove generated files, and patch required files.
-
-if ! [ -f build/autogen-5.18.16.tar.xz ]; then
-	echo "=== Downloading tarball ==="
-	curl https://mirrors.kernel.org/gnu/autogen/rel5.18.16/autogen-5.18.16.tar.xz -o build/autogen-5.18.16.tar.xz
-fi
-
-if ! [ -d build/src ]; then
-	echo "=== Retrieving missing files from repository ==="
-	cd build
-	curl https://git.savannah.gnu.org/cgit/autogen.git/snapshot/autogen-5.18.16.tar.gz -o autogen-5.18.16.tar.gz
-	tar -xf autogen-5.18.16.tar.gz
-	mv autogen-5.18.16 src
-	cd src
-	rm -f add-on/char-mapper/cm.tar
-	cd ../..
+main() {
+	download_release_tarball
+	download_snapshot_tarball
+	prepare_tarball
+	bootstrap_columns
+	bootstrap_getdefs
+	bootstrap_autogen
+	bootstrap_tpl_config
+	rebuild_autogen
+}
+
+download_release_tarball() {
+	## Get the tarball
+	if ! [ -f build/autogen-5.18.16.tar.xz ]; then
+		echo "=== Downloading tarball ==="
+		mkdir build
```

Comment:
I'd still run both scripts repeatedly (or first one and then the other). So the `-p` needs to stay.